alphadia/transferlearning/train.py

```diff
@@ -896,3 +897,166 @@ def finetune_charge(self, psm_df: pd.DataFrame) -> pd.DataFrame:
         metrics = test_metric_manager.get_stats()
 
         return metrics
+
+    def _test_ccs(
+        self,
+        epoch: int,
+        epoch_loss: float,
+        test_df: pd.DataFrame,
+        metric_accumulator: MetricManager,
+        data_split: str,
+    ) -> bool:
+        """
+        Test the CCS model using the PSM dataframe and accumulate both the training loss and test metrics.
+
+        Parameters
+        ----------
+        epoch : int
+            The current epoch number.
+        epoch_loss : float
+            The train loss value of the current epoch.
+        test_df : pd.DataFrame
+            The PSM dataframe.
+        metric_accumulator : MetricManager
+            The metric manager object.
+        data_split : str
+            The dataset label to test on. e.g. "validation", "train"
+        Returns
+        -------
+        bool
+            Whether to continue training or not based on the early stopping criteria.
+        """
+        continue_training = True
+        if epoch % self.settings["test_interval"] == 0:
+            self.ccs_model.model.eval()
+
+            pred = self.ccs_model.predict(test_df)
+
+            test_input = {
+                "predicted": pred["ccs_pred"].values,
+                "target": test_df["ccs"].values,
+            }
+            val_metrics = metric_accumulator.calculate_test_metric(
+                test_input, epoch, data_split=data_split, property_name="ccs"
+            )
+            if epoch != -1 and data_split == "validation":
```

Comment:
does the check for `epoch != -1` make sense here?
if `epoch = -1`, we will be here only if `self.settings["test_interval"] = 1`